internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/metadata.go

```diff
@@ -296,6 +299,10 @@ func (n *Nocloud) acquireConfig(ctx context.Context, r state.State) (metadataNet
 		metadata.Hostname = hostname
 	}
 
+	if instanceID != "" {
```

Comment:
should we overwrite here `metadata.InstanceID` if it's non-empty (something was downloaded?)

Reply:
Goot question. 

It can only be set using the special string format: `-smbios type=1,serial=ds=nocloud;h=my-host-name;i=id-1234`. By default, this value is empty.

I know that the cloud (Proxmox) sets a randomly generated instance-id in the metadata file. This value cannot be reliably used in the future by code, as there is no mapping to a real ID, UUID, or other identifiers.

This PR aims to override the randomly generated Proxmox value.

PS. The hostname in smbios has priority too.

Thanks

Reply:
sounds good, thanks for clarifying!